File: pinyougouparent/pinyougou-web-user/src/main/webapp/js/service/addressService.js
//服务层
app.service('addressService',function($http){
    //读取列表数据绑定到表单中
    this.showAddress=function(){
        return $http.get('../address/findListByLoginUser.do');
    }

    this.findProvinces = function(){
        return $http.get("../address/findProvinces.do");
    }
    this.findCities = function(provinceId){
        return $http.get("../address/findCities.do?provinceId="+provinceId);
    }
    this.findAreas = function(cityId){
        return $http.get("../address/findAreas.do?cityId="+cityId);
    }

    this.add=function(address){
        return  $http.post('../address/add.do',address );
    }

    this.delete=function(id){
        return  $http.get('../address/delete.do?id='+id);
    }
});

File: pinyougouparent/pinyougou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String name);

    List<Provinces> findProvinces();

    List<Cities> findCities(Long provinceId);

    List<Areas> findAreas(Long cityId);

    void add(Address address,String name);

    void delete(Long id);
}
